### ISSUE TITLE
CORS Can burn in hell

### ISSUE THREAD

**Issue by TehWardy** [dotnet/aspnetcore#24577]
Every time someone patches a browser CORS issues come back.
Either this standard is an ever moving target or the implementation in aspnet core just doesn't work or I have no idea how to configure it.

I've tried just about every combination of documented configurations for it that I can find and it still comes back every couple of weeks for seemingly no reason other than aspnet core decided that it was going to return the headers the browser wants "sometimes".

How do I setup CORS to allow any request anywhere in the from any caller under any condition so that I can be done with this hell?

Here's what I currently have in my startup and it's not working ...

```
public void ConfigureServices(IServiceCollection services)
{
    ...
    services.AddMvcCore(options => options.EnableEndpointRouting = false)
        .AddDataAnnotations()
        .AddJsonOptions(options => options.JsonSerializerOptions.IgnoreNullValues = true)
        .AddCors(options => options.AddDefaultPolicy(builder =>
        {
            builder.AllowAnyHeader();
            builder.AllowAnyMethod();
            builder.SetIsOriginAllowed(origin => true);
            builder.AllowCredentials();
        }));
    ...
{

public void Configure(IApplicationBuilder app)
{
    ...
    app.UseCors(builder =>
    {
        builder.AllowAnyHeader();
        builder.AllowAnyMethod();
        builder.SetIsOriginAllowed(origin => true);
        builder.AllowCredentials();
    });
    ...
}
```

Todays exception of choice is ...

> Access to XMLHttpRequest at 'https://...' from origin 'https://...' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The value of the 'Access-Control-Allow-Credentials' header in the response is '' which must be 'true' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.

.. on "some" calls.

**Comment by mkArtakMSFT** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide a minimalistic repro project (ideally a GitHub repo) that illustrates the problem.

**Comment by TehWardy** [comment]
The example above is my CORS configuration that causes the issue.
It seems that "in some cases" which chrome / edge seem to fluctuate between liking and not liking it "mostly works".
I'm thinking I may just give us and re-engineer my applications to do less / no CORS calls at all as the technical leads in this area seem intent on making my life hell.

I wish there was a way to configure aspnet core to say "I don't care what the request is for, if it's an options call just return a valid CORS response and let the actual call handle it properly".
I know this is anti what CORS is supposed to achieve but IMO the whole standard is flawed since it relies on the calling clients information being trusted in order to make an informed security response.

My issue is that I build a platform that all our clients share and they are from all over the world and operate on multiple domains so we assess the users ability to do anything based on session tokens not what site they may or may not be using at the time they call in to our API domain.

If you guys think this is a browser issue at this point I would fully understand, feel free to close this I just wanted to voice my distress that CORS is even a thing and worse that i'm forced to use it.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
CORS configuration is really hard and it can indeed be frustrating. While we can try to improve this area slightly to help here, it's not a priority for us at the moment. Given that we haven't received much similar feedback we're closing this issue.
